contracts/nois-airdrop/src/msg.rs

```diff
@@ -0,0 +1,80 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{HexBinary, Timestamp, Uint128};
+use nois::NoisCallback;
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    /// manager if none set to info.sender.
+    pub manager: String,
+    /// Address of the Nois proxy contract
+    pub nois_proxy: String,
+}
+
+#[cw_serde]
+pub enum ExecuteMsg {
+    UpdateConfig {
+        manager: Option<String>,
+    },
+    RegisterMerkleRoot {
+        /// MerkleRoot is hex-encoded merkle root.
+        merkle_root: HexBinary,
+    },
+    // This will trigger fetching the unpredictable random beacon that will serve to give the randdrop to only a subset of accounts
+    RandDrop {
+        random_beacon_after: Timestamp,
```

Comment:
Is there a way to to write a safer API here? Like X hours from now?

Using the `after` value directly is very unsafe. It's not checked at all at the moment. If you miss some 0 in the number, you kill the airdrop.